packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -412,6 +422,7 @@ const serviceHeaderDefaults = {
 class ReportData {
   report!: Report;
   size!: number;
+  traceCache: Map<TraceCacheKey, boolean> = new Map<TraceCacheKey, boolean>();
```

Comment:
I don't see where you write to this cache

Reply:
Oh yea. Thanks. Didn't write it yet. Added after trace encoding.

Reply:
[JS `Map` bases key equality on the sameValueZero algorithm](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)
[sameValueZero is almost the same as same-value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#Same-value-zero_equality)
[same-value is the same as Object.is](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#Same-value_equality)
[Object.is doesn't do any sort of recursive structural comparison on Objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is)

or in other words

```
> x = new Map(); x.set({}, 'foo'); x.get({})
undefined
```

JSON.stringify is likely your friend